index.js

```diff
@@ -3,26 +3,101 @@
 'use strict';
 var spawn = require('child_process').spawn;
 
-function potentialExit (childCmd, code) {
+var sh, shFlag, children, args, wait, cmds, verbose, i ,len;
+// parsing argv
+cmds = [];
+args = process.argv.slice(2);
+for (i = 0, len = args.length; i < len; i++) {
+    if (args[i][0] === '-') {
+        switch (args[i]) {
+            case '-w':
+            case '--wait':
+                wait = true;
+                break;
+            case '-v':
+            case '--verbose':
+                verbose = true;
+                break;
+            case '-h':
+            case '--help':
+                console.log('-h, --help         output usage information');
+                console.log('-v, --verbose      verbose logging')
+                console.log('-w, --wait         will not close silbling processes on error')
+                process.exit();
+                break;
+        }
+    } else {
+        cmds.push(args[i]);
+    }
+}
+
+// called on close of a child process
+function childClose (code) {
+    var i, len;
     code = code? (code.code || code) : code;
-    if (code > 0) {
-        console.error('`' + childCmd + '` failed with exit code ' + code);
-        process.exit(code);
+    if (verbose) {
+        if (code > 0) {
+            console.error('`' + this.cmd + '` failed with exit code ' + code);
+        } else {
+            console.log('`' + this.cmd + '` ended successfully');
+        }
+    }
+    if (code > 0 && !wait) close();
+    status();
+}
+
+function status () {
+    if (verbose) {
+        var i, len;
+        console.log('\n');
+        console.log('### Status ###');
+        for (i = 0, len = children.length; i < len; i++) {
+            if (children[i].exitCode === null) {
+                console.log('`' + children[i].cmd + '` is still running');
+            } else if (children[i].exitCode > 0) {
+                console.log('`' + children[i].cmd + '` errored');
+            } else {
+                console.log('`' + children[i].cmd + '` finished');
+            }
+        }
+        console.log('\n');
     }
 }
-var sh = 'sh';
-var shFlag = '-c';
 
+// closes all children and the process
+function close () {
+    var i, len;
+    for (i = 0, len = children.length; i < len; i++) {
+        if (!children[i].exitCode) {
+            children[i].removeAllListeners('close');
+            children[i].kill('SIGINT');
+            if (verbose) console.log('`' + children[i].cmd + '` will now be closed');
+        }
+    }
+    process.exit();
```

Comment:
There is a breaking change here. Before, parallelshell would exit with a non-zero exit code if the child process did, now it does not. This close function should ideally find the first non-zero exit code and exit with that (or if all exit codes are 0, of course, then exit with that). 

Simply refactor this function as follows:

```js
function close () {
    var code = 0;
    for (var i = 0, len = children.length; i < len; i++) {
        if (!children[i].exitCode) {
            children[i].removeAllListeners('close');
            children[i].kill('SIGINT');
            if (verbose) console.log('`' + children[i].cmd + '` will now be closed');
        } else {
            code = children[i].exitCode;
        }
    }
    process.exit(code);
}
```